### PR TITLE
Request if chaos mode was enabled

### PR DESCRIPTION
**Pull request by rasom**
Please review only last two commits.

The warning is also added in this PR

Screen Shot 2019-03-19 at 16 59 50

Record activated:
```
{
  "Status": 0,
  "TC": false,
  "RD": true,
  "RA": true,
  "AD": false,
  "CD": false,
  "Question": [
    {
      "name": "chaos-unicorn-day-test.status.im.",
      "type": 16
    }
  ],
  "Answer": [
    {
      "name": "chaos-unicorn-day-test.status.im.",
      "type": 16,
      "TTL": 273,
      "data": "\"DO IT LIVE!\""
    }
  ]
}
```

Record doesn't exist:
```
{
  "Status": 3,
  "TC": false,
  "RD": true,
  "RA": true,
  "AD": false,
  "CD": false,
  "Question": [
    {
      "name": "chaos-unicorn-day-test-foo.status.im.",
      "type": 16
    }
  ],
  "Authority": [
    {
      "name": "status.im.",
      "type": 6,
      "TTL": 3584,
      "data": "grace.ns.cloudflare.com. dns.cloudflare.com. 2030466987 10000 2400 604800 3600"
    }
  ]
}
```

### CODE REVIEW

**Comment by mandrigin** [comment]
@rasom looks good but at least an example of JSON for chaos on/off would be nice to have in the description of this PR.

**Comment by rasom** [comment]
@mandrigin ok

**Comment by rasom** [comment]
@mandrigin added

File: src/status_im/models/transactions.cljs [hunk 1]
```diff
@@ -177,12 +178,17 @@
...
-    (str "https://" network-subdomain ".etherscan.io/api?module=account&action=txlist&address=0x"
-         account "&startblock=0&endblock=99999999&sort=desc&apikey=" etherscan-api-key "&q=json")))
+(defn- get-transaction-url
+  ([chain account] (get-transaction-url chain account false))
+  ([chain account chaos-mode?]
+   {:pre  [(keyword? chain) (string? account)]
+    :post [(string? %)]}
+   (let [network-subdomain (get-api-network-subdomain chain)]
+     (if chaos-mode?
+       "http://httpstat.us/500"
```
Comment (mandrigin) [inline src/status_im/models/transactions.cljs]: 👍

File: src/status_im/chaos_mode/core.cljs [hunk 2]
```diff
@@ -0,0 +1,44 @@
+(ns status-im.chaos-mode.core
+  (:require [status-im.utils.fx :as fx]
+            [re-frame.core :as re-frame]
+            [status-im.utils.utils :as utils]
+            [status-im.utils.http :as http]
+            [status-im.utils.types :as types]))
+
+(def interval-5m (* 5 60 1000))
+(defonce interval-id (atom nil))
+(def url "https://cloudflare-dns.com/dns-query?name=chaos-unicorn-day-test.status.im&type=TXT")
```
Comment (mandrigin) [inline src/status_im/chaos_mode/core.cljs]: this domain name will change, right? cc @jakubgs

**Review by mandrigin — APPROVED** [review]

File: src/status_im/chaos_mode/core.cljs (hunk 2 above)
Comment (rasom) [reply src/status_im/chaos_mode/core.cljs]: yep, most likely. I will change it before merging

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/chaos_mode/core.cljs (hunk 2 above)
Comment (jakubgs) [reply src/status_im/chaos_mode/core.cljs]: Yeah, i assume the real one we will be using on that day will be `chaos-unicorn-day.status.im`, without the `-test`.

**Comment by annadanchenko** [comment]
@rasom issue 1. If user signed in when no Internet connection available then instead of unicorn warning there is popup "TypeError: Network request failed". If connection is available after this then no popup and no Chaos mode turned on... So users who had troubles with connection will not get this warning

**Comment by rasom** [comment]
@annadanchenko on it

**Comment by anna02test** [comment]
@rasom it's from Anna's test gh account
issue 2 on windows build 4. If create an account and log in then popup is shown "status_im.react_native.js_dependencies.background_timer.setInterval is not a function"

PRETTY PRINTED EXCEPTION

***
NAME: "TypeError"
MESSAGE: status_im.react_native.js_dependencies.background_timer.setInterval is not a function

TypeError: status_im.react_native.js_dependencies.background_timer.setInterval is not a function
    at Object.status_im.utils.utils.set_interval (evalmachine.<anonymous>:11:3735414)
    at G__33127_33129 (evalmachine.<anonymous>:11:6578543)
    at evalmachine.<anonymous>:11:1682777
    at Object.re_frame.interceptor.invoke_interceptor_fn (evalmachine.<anonymous>:11:1638018)
    at Object.re_frame.interceptor.invoke_interceptors (evalmachine.<anonymous>:11:1638302)
    at Object.re_frame.interceptor.execute (evalmachine.<anonymous>:11:1640674)
    at Object.re_frame.events.handle (evalmachine.<anonymous>:11:1649178)
    at re_frame.router.EventQueue.re_frame$router$IEventQueue$_process_1st_event_in_queue$arity$1 (evalmachine.<anonymous>:11:1676264)
    at re_frame.router.EventQueue.re_frame$router$IEventQueue$_run_queue$arity$1 (evalmachine.<anonymous>:11:1661192)
    at evalmachine.<anonymous>:11:1664305

***
17:07:34.930 - D - [jsserver] - [status-im.utils.handlers:34] - Handling re-frame event:  :accounts.ui/chaos-mode-switched
[Status.log](https://github.com/status-im/status-react/files/2989103/Status.log)

**Comment by rasom** [comment]
@annadanchenko should be fixed in the next build

**Comment by annadanchenko** [comment]
issue 3: on windows build 4 - if open About box then app crashes with `TypeError: re-matches must match against string"

[Status (3).log](https://github.com/status-im/status-react/files/2989126/Status.3.log)

errrrr

**Comment by rasom** [comment]
What exactly is "About box"?

**Comment by annadanchenko** [comment]
> What exactly is "About box"?

Profile -> About

**Comment by rasom** [comment]
@annadanchenko the second issue is fixed

I can't reproduce the third one - I don't have windows. Also I'm quite puzzled how it can be related to this PR...

**Comment by annadanchenko** [comment]
@rasom I can't repro issue 2 (as it's fixed) and issue 3 anymore... will continue testing new builds tomorrow

**Comment by churik** [comment]
@annadanchenko tested on Linux 18.04.

After creating account faced with https://github.com/status-im/status-react/issues/7776 (already in `develop`). But can use normally with an existing account (can login \ logout, join public chat, switch mailserver, open About, Devices in profile, backup recovery phrase)
So looks good to me.

**Comment by annadanchenko** [comment]
@rasom looks ok on Windows, iOS and Android to me. Also tested that switching off dns record leads to Chaos mode being turned off. Triggered autotests (with switched off DNS record) to see if there are no regressions. After they are done - ok to merge. 

Don't forget to change url before merge

**Comment by rasom** [comment]
@annadanchenko ok

**Comment by annadanchenko** [comment]
sorry, will trigger them again but with tests from this branch not from develop that got some changes already...

**Comment by annadanchenko** [comment]
@rasom ok to merge. 1 failing test is because faucet does not work atm.

**Pull request merged**